src/libmongoc/tests/unified/operation.c

```diff
@@ -649,6 +652,65 @@ operation_count_documents (test_t *test,
    return ret;
 }
 
+static bool
+operation_create_find_cursor (test_t *test,
+                              operation_t *op,
+                              result_t *result,
+                              bson_error_t *error)
+{
+   bool ret = false;
+   mongoc_collection_t *coll = NULL;
+   bson_parser_t *parser = NULL;
+   mongoc_cursor_t *cursor = NULL;
+   bson_t *filter = NULL;
+   bson_t *opts = NULL;
+   const bson_t *op_reply = NULL;
+   bson_error_t op_error = {0};
+
+   parser = bson_parser_new ();
+   bson_parser_allow_extra (parser, true);
+   bson_parser_doc (parser, "filter", &filter);
+   if (!bson_parser_parse (parser, op->arguments, error)) {
+      goto done;
+   }
+
+   coll = entity_map_get_collection (test->entity_map, op->object, error);
+   if (!coll) {
+      goto done;
+   }
+
+   opts = bson_copy (bson_parser_get_extra (parser));
+   if (op->session) {
+      if (!mongoc_client_session_append (op->session, opts, error)) {
+         goto done;
+      }
+   }
+
+   cursor = mongoc_collection_find_with_opts (
+      coll, filter, opts, NULL /* read prefs */);
+
+
+   if (!op->save_result_as_entity) {
+      test_set_error (error,
+                      "unexpected createFindCursor does not save result");
```

Comment:
[createFindCursor](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#createfindcursor) doesn't actually require `saveResultAsEntity`, so I don't think this is necessary. Did you do this to avoid a potential leak if the entity wasn't going to be saved?

I think without `saveResultAsEntity`, you could just destroy the cursor immediately. A test might still want to do this to assert that a `find` command is issued.